cmd/sys-mgmt/main.go

```diff
@@ -0,0 +1,105 @@
+	"github.com/edgexfoundry/edgex-go/pkg/clients/logging"
+	"github.com/edgexfoundry/edgex-go/internal/system/agent"
+)
+
+var bootTimeout int = 30000 //Once we start the V2 configuration rework, this will be config driven
+
+func main() {
+
+	// TODO: 1.  What kind of discoverability scenarios should be considered?
+	// TODO: 2.  What kind of 'interaction' scenarios (involving Vault and Kong) should be considered?
+	// TODO: 3.  Allow the agent/APIs to be turned off (or not used)...
+	// TODO: 4.  Each service containing the SM API must have a configuration setting that turns off, protects or otherwise causes the SM API on the services to no-op.
+	// TODO: 5.  When (if!) Consul is running, the SMA could request Consul to provide a catalog of registered services. To remove reliance on Consul, the SMA will be provided configuration (a manifest) that specifies the services it is to manage (Thus, the SMA can bootstrap all of EdgeX!)
+	// TODO: 6.  Add a bootstrap property akin to –consul (e.g. –sma) to indicate directive for fetching the configuration from the local file system.
+	// TODO: 7.  Effectively, the SMA can either (1) get configuration from Consul, or (2) get configuration from the local file system. This will be a manifest configuration file.
+	// TODO: 8.  This file will boostrap the SMA. Different versions of the file may exist depending on how/where EdgeX is deployed (Docker v. Snappy, Windows v. Linux, etc.).
+	// TODO: 9.  System Of Record (SOR) for configuration: The SMA must request configuration information from the _service_ itself!
+	// TODO: 10. And BTW, what of the SMA's managing 3rd party infrastructure services (MongoDB, Consul, Kong, etc.) that will _not_ adhere to EdgeX guidelines?
+
+	start := time.Now()
+	var useConsul bool
+	var useProfile string
+
+	flag.BoolVar(&useConsul, "consul", false, "Indicates the service should use consul.")
+	flag.BoolVar(&useConsul, "c", false, "Indicates the service should use consul.")
+	flag.StringVar(&useProfile, "profile", "", "Specify a profile other than default.")
+	flag.StringVar(&useProfile, "p", "", "Specify a profile other than default.")
+	flag.Usage = usage.HelpCallback
+	flag.Parse()
+
+	params := startup.BootParams{UseConsul: useConsul, UseProfile: useProfile, BootTimeout: bootTimeout}
+	startup.Bootstrap(params, agent.Retry, logBeforeInit)
+
+	ok := agent.Init()
+	if !ok {
+		fmt.Printf("Service bootstrap failed!")
+		return
+	}
+
+	fmt.Printf("Service dependencies resolved...")
+	fmt.Printf(fmt.Sprintf("Starting %s %s ", internal.SystemManagementAgentServiceKey, edgex.Version))
+
+	http.TimeoutHandler(nil, time.Millisecond*time.Duration(agent.Configuration.ServiceTimeout), "Request timed out")
+	fmt.Printf(agent.Configuration.AppOpenMsg, "")
+
+	errs := make(chan error, 2)
+	listenForInterrupt(errs)
+	startHttpServer(errs, agent.Configuration.ServicePort)
+
+	// Time it took to start service
+	fmt.Printf("Service started in: "+time.Since(start).String(), "")
```

Comment:
No `printf` on stdout - use logger.

Reply:
@akramtexas To @drasko's point, follow the patterns available in other services where main uses LoggingClient to log. In cases where you have to log before bootstrap is complete (or it has failed) see `func logBeforeInit(err error)`